drivers/sensor/smart_battery/smart_battery.c

```diff
@@ -0,0 +1,284 @@
+		if (status < 0) {
+			LOG_ERR("Failed to read full charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		status = smart_battery_command_reg_read(
+			dev, SMART_BATTERY_COMMAND_REM_CAPACITY,
+			&smart_battery->remaining_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read remaining charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		status = smart_battery_command_reg_read(dev,
+						  SMART_BATTERY_COMMAND_NOM_CAPACITY,
+						  &smart_battery->nom_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read nominal available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		status =
+			smart_battery_command_reg_read(dev,
+						 SMART_BATTERY_COMMAND_FULL_CAPACITY,
+						 &smart_battery->full_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		status = smart_battery_command_reg_read(dev, SMART_BATTERY_COMMAND_AVG_TIME2EMPTY,
+						  &smart_battery->time_to_empty);
+
+		smart_battery->time_to_empty = (smart_battery->time_to_empty) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		status = smart_battery_command_reg_read(dev, SMART_BATTERY_COMMAND_AVG_TIME2FULL,
+						  &smart_battery->time_to_full);
+
+		smart_battery->time_to_full = (smart_battery->time_to_full) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_CYCLE_COUNT:
+		status = smart_battery_command_reg_read(dev, SMART_BATTERY_COMMAND_CYCLE_COUNT,
+						  &smart_battery->cycle_count);
+
+		smart_battery->cycle_count = (smart_battery->cycle_count) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief initialise the fuel gauge
+ *
+ * @return 0 for success
+ */
+static int smartbattery_gauge_init(const struct device *dev)
+{
+	struct smartbattery_data *smart_battery = DEV_DATA(dev);
+	const struct smartbattery_config *const config = DEV_CFG(dev);
+
+	smart_battery->i2c = device_get_binding(config->bus_name);
+	if (smart_battery->i2c == NULL) {
+		LOG_ERR("Could not get pointer to %s device.",
+			config->bus_name);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api smartbattery_battery_driver_api = {
+	.sample_fetch = smartbattery_sample_fetch,
+	.channel_get = smartbattery_channel_get,
+};
+
+#define SMART_BATTERY_INIT(index) \
+	static struct smartbattery_data smartbattery_driver_##index; \
+\
+	static const struct smartbattery_config smartbattery_config_##index = { \
+		.bus_name = DT_INST_BUS_LABEL(index), \
```

Comment:
Seems that you forgot to change to Is there an alias for .i2c_dev= DEVICE_DT_GET(DT_INST_BUS(index))

Reply:
true, will change that